#' Title
#'
#' @param species
#' @param collicin
#' @param annotationDir
#' @param genomeDir
#' @param outDir
#' @return
#' @export

extract.all.genes.annotationbased <- function(species,genemlst,annotationDir,genomeDir,outDir)
{
  library(reutils)
  library(ape)
  library(seqinr)
  library(Biostrings)
  library(dplyr)

  nspecies <- length(species)
  ngenes <- length(genemlst)


  for(i in 1:nspecies)
  {
    dir.create(paste0(outDir,species[i]))
    for(j in 1:ngenes)
    {
      extract.1.gene.annotationbased(selectedspecies=species[i],selectedgene=genemlst[j],annotationDir=annotationDir,genomeDir=genomeDir,outDir=outDir)
    }
    print(i)
  }

  fasta.list=c()
  for (i in 1:nspecies) {
    fasta.list.newspecies <- list.files(paste0(outDir,species[i],'/'),full.names = T)
    fasta.list.newspecies <- fasta.list.newspecies[grep('fasta',fasta.list.newspecies)]
    fasta.list <- c(fasta.list,fasta.list.newspecies)
  }

  genename <- basename(fasta.list)
  for(i in 1:length(genename))
  {
    current.sequence <- readDNAStringSet(fasta.list[grep(genename[i],fasta.list)])
    writeXStringSet(current.sequence,paste0(outDir,genename[i]))
  }


}
